src/Cake.SqlServer/Backup/RestoreSqlBackupImpl.cs

```diff
@@ -152,6 +178,25 @@ internal static String GetDefaultDataPath(SqlConnection connection)
             return defaultPath;
         }
 
+        internal static OS GetSqlServerOS(SqlConnection connection)
+        {
+            var sql = "select @@version as version";
+
+            var version = ReadSingleString(sql, "version", connection).ToLower();
+
+            if (version.Contains("windows"))
+            {
+                return OS.Windows;
```

Comment:
Why have an extra step of returning OS and extra `case` statement, when you can just return a separator char after finding out a version?
I'd flatten these 2 functions into 1 and remove the new enum as surplus

Reply:
I thought of doing that, but then maybe something else platform specific will pop up and we will need to react on that. If you insist, I will just move it into one function.

Reply:
I think it is an overkill here. YAGNI. Nothing particular complex about this code, so it'll be easy to restore/refactor when need arises. I say simplify it.
Also, do you have SQL Server on Linux handy to test? I don't really have a way to check if this is working.

Reply:
I'm testing with a docker container, as this is going to be my main use case. https://hub.docker.com/r/microsoft/mssql-server-linux/
The problem before that was that SQL Server was complaining about bad paths, etc. I will clean up the code and update the PR :)